components/camel-lra/src/test/java/org/apache/camel/service/lra/LRAClientTest.java

```diff
@@ -56,11 +62,92 @@ void testCannotCreateLRAClientWithoutHttpClient() throws Exception {
                 "no client should result in IllegalArgumentException");
     }
 
+    @DisplayName("Tests whether LRAClient is calling prepareRequest with exchange from newLRA()")
+    @Test
+    void testCallsPrepareRequestWithExchangeInNewLra() {
+        LRASagaService sagaService = new LRASagaService();
+        applyMockProperties(sagaService);
+        LRAClient client = new LRAClient(sagaService) {
+            protected HttpRequest.Builder prepareRequest(URI uri, Exchange exchange) {
+                throw new ExchangeRuntimeException(exchange);
+            }
+        };
+        Exchange exchange = Mockito.mock(Exchange.class);
+        try {
+            client.newLRA(exchange);
+            Assertions.fail("there should have been the special testing exception ExchangeRuntimeException");
+        } catch (ExchangeRuntimeException ex) {
+            Assertions.assertSame(exchange, ex.get());
+        }
```

Comment:
This is usually better done using `assertThrows`.

Reply:
You are so true. I will immediately rework on it.